Streets disappear from the map in some cases

In darkmode in chrome / androidy streets are not visible on the map.

The question is if we want a special darkmode map style or just force the original theme in darkmode for the map?

![Screenshot_20240601-205417](https://github.com/zbycz/osmapp/assets/1087128/68b46aff-494a-4f4d-ae90-a0064ee7bca3)

See also #134

This is the error log, where the last warning appears just in the moment when the map disappears (I close the popup):

I think this is the relevant one.
```
helpers.ts:62 WARNING: Too many active WebGL contexts. Oldest context will be lost.
```

Code line:

[      const context = canvas.getContext('webgl2') || canvas.getContext('webgl');
](https://github.com/zbycz/osmapp/blob/7137cba9c507300ccda538d12d25ae83d439e99b/src/components/Map/helpers.ts#L62)


<details>

```
instrument.js:109 Geolocation support is not available so the GeolocateControl will be disabled.
(anonymous) @ instrument.js:109
Show 1 more frame
Show less
instrument.js:109 Image "sports_centre_11" could not be loaded. Please make sure you have added the image with map.addImage() or a "sprite" property in your style. You can provide missing images by listening for the "styleimagemissing" map event.
(anonymous) @ instrument.js:109
B @ maplibre-gl.js:31
_notify @ maplibre-gl.js:35
getImages @ maplibre-gl.js:35
getImages @ maplibre-gl.js:35
processTask @ maplibre-gl.js:31
receive @ maplibre-gl.js:31
o @ helpers.js:74
Show 8 more frames
Show less
instrument.js:109 Image "doctors_11" could not be loaded. Please make sure you have added the image with map.addImage() or a "sprite" property in your style. You can provide missing images by listening for the "styleimagemissing" map event.
(anonymous) @ instrument.js:109
B @ maplibre-gl.js:31
_notify @ maplibre-gl.js:35
getImages @ maplibre-gl.js:35
getImages @ maplibre-gl.js:35
processTask @ maplibre-gl.js:31
receive @ maplibre-gl.js:31
o @ helpers.js:74
Show 8 more frames
Show less
instrument.js:109 Image "office_11" could not be loaded. Please make sure you have added the image with map.addImage() or a "sprite" property in your style. You can provide missing images by listening for the "styleimagemissing" map event.
(anonymous) @ instrument.js:109
B @ maplibre-gl.js:31
_notify @ maplibre-gl.js:35
getImages @ maplibre-gl.js:35
getImages @ maplibre-gl.js:35
processTask @ maplibre-gl.js:31
receive @ maplibre-gl.js:31
o @ helpers.js:74
Show 8 more frames
Show less
instrument.js:109 Image "butcher_11" could not be loaded. Please make sure you have added the image with map.addImage() or a "sprite" property in your style. You can provide missing images by listening for the "styleimagemissing" map event.
(anonymous) @ instrument.js:109
B @ maplibre-gl.js:31
_notify @ maplibre-gl.js:35
getImages @ maplibre-gl.js:35
getImages @ maplibre-gl.js:35
processTask @ maplibre-gl.js:31
receive @ maplibre-gl.js:31
o @ helpers.js:74
Show 8 more frames
Show less
instrument.js:109 Image "water_park_11" could not be loaded. Please make sure you have added the image with map.addImage() or a "sprite" property in your style. You can provide missing images by listening for the "styleimagemissing" map event.
(anonymous) @ instrument.js:109
B @ maplibre-gl.js:31
_notify @ maplibre-gl.js:35
getImages @ maplibre-gl.js:35
getImages @ maplibre-gl.js:35
processTask @ maplibre-gl.js:31
receive @ maplibre-gl.js:31
o @ helpers.js:74
Show 8 more frames
Show less
useOnMapClicked.tsx:53 clicked map feature (id=8623283):  ow {type: 'Feature', _vectorTileFeature: n0, properties: {…}, id: 8623283, layer: {…}, …}
instrument.js:109 Image "basin_11" could not be loaded. Please make sure you have added the image with map.addImage() or a "sprite" property in your style. You can provide missing images by listening for the "styleimagemissing" map event.
(anonymous) @ instrument.js:109
B @ maplibre-gl.js:31
_notify @ maplibre-gl.js:35
getImages @ maplibre-gl.js:35
getImages @ maplibre-gl.js:35
processTask @ maplibre-gl.js:31
receive @ maplibre-gl.js:31
o @ helpers.js:74
Show 8 more frames
Show less
useOnMapClicked.tsx:53 clicked map feature (id=3002749620):  ow {type: 'Feature', _vectorTileFeature: n0, properties: {…}, id: 3002749620, layer: {…}, …}
useFeatureMarker.tsx:21 Error: The layer 'climbing-3-routes-circle' does not exist in the map's style and cannot be queried for features.
    at eX.queryRenderedFeatures (maplibre-gl.js:35:121676)
    at rR.Map.queryRenderedFeatures (maplibre-gl.js:35:369008)
    at T (useFeatureMarker.tsx:21:23)
    at useFeatureMarker.tsx:45:5
    at helpers.tsx:34:20
    at uI (react-dom.production.min.js:244:322)
    at oU (react-dom.production.min.js:286:88)
    at react-dom.production.min.js:283:386
    at oI (react-dom.production.min.js:282:143)
    at oC (react-dom.production.min.js:273:439)
    at r4 (react-dom.production.min.js:127:100)
    at react-dom.production.min.js:267:262
(anonymous) @ instrument.js:109
fire @ maplibre-gl.js:31
queryRenderedFeatures @ maplibre-gl.js:35
queryRenderedFeatures @ maplibre-gl.js:35
T @ useFeatureMarker.tsx:21
(anonymous) @ useFeatureMarker.tsx:45
(anonymous) @ helpers.tsx:34
uI @ react-dom.production.min.js:244
oU @ react-dom.production.min.js:286
(anonymous) @ react-dom.production.min.js:283
oI @ react-dom.production.min.js:282
oC @ react-dom.production.min.js:273
r4 @ react-dom.production.min.js:127
(anonymous) @ react-dom.production.min.js:267
Show 11 more frames
Show less
helpers.ts:111 getInititalFeature(n300274962): 560ms [osm] + 0ms [ssr img]
useFeatureMarker.tsx:21 Error: The layer 'climbing-3-routes-circle' does not exist in the map's style and cannot be queried for features.
    at eX.queryRenderedFeatures (maplibre-gl.js:35:121676)
    at rR.Map.queryRenderedFeatures (maplibre-gl.js:35:369008)
    at T (useFeatureMarker.tsx:21:23)
    at useFeatureMarker.tsx:36:5
    at helpers.tsx:34:20
    at uI (react-dom.production.min.js:244:322)
    at oU (react-dom.production.min.js:286:88)
    at o (react-dom.production.min.js:282:391)
    at x (scheduler.production.min.js:13:197)
    at MessagePort.T (scheduler.production.min.js:14:126)
(anonymous) @ instrument.js:109
fire @ maplibre-gl.js:31
queryRenderedFeatures @ maplibre-gl.js:35
queryRenderedFeatures @ maplibre-gl.js:35
T @ useFeatureMarker.tsx:21
(anonymous) @ useFeatureMarker.tsx:36
(anonymous) @ helpers.tsx:34
uI @ react-dom.production.min.js:244
oU @ react-dom.production.min.js:286
o @ react-dom.production.min.js:282
x @ scheduler.production.min.js:13
T @ scheduler.production.min.js:14
Show 9 more frames
Show less
useFeatureMarker.tsx:21 Error: The layer 'climbing-3-routes-circle' does not exist in the map's style and cannot be queried for features.
    at eX.queryRenderedFeatures (maplibre-gl.js:35:121676)
    at rR.Map.queryRenderedFeatures (maplibre-gl.js:35:369008)
    at T (useFeatureMarker.tsx:21:23)
    at useFeatureMarker.tsx:45:5
    at helpers.tsx:34:20
    at uI (react-dom.production.min.js:244:322)
    at oU (react-dom.production.min.js:286:88)
    at o (react-dom.production.min.js:282:391)
    at x (scheduler.production.min.js:13:197)
    at MessagePort.T (scheduler.production.min.js:14:126)
(anonymous) @ instrument.js:109
fire @ maplibre-gl.js:31
queryRenderedFeatures @ maplibre-gl.js:35
queryRenderedFeatures @ maplibre-gl.js:35
T @ useFeatureMarker.tsx:21
(anonymous) @ useFeatureMarker.tsx:45
(anonymous) @ helpers.tsx:34
uI @ react-dom.production.min.js:244
oU @ react-dom.production.min.js:286
o @ react-dom.production.min.js:282
x @ scheduler.production.min.js:13
T @ scheduler.production.min.js:14
Show 9 more frames
Show less
useFeatureMarker.tsx:21 Error: The layer 'climbing-3-routes-circle' does not exist in the map's style and cannot be queried for features.
    at eX.queryRenderedFeatures (maplibre-gl.js:35:121676)
    at rR.Map.queryRenderedFeatures (maplibre-gl.js:35:369008)
    at T (useFeatureMarker.tsx:21:23)
    at useFeatureMarker.tsx:36:5
    at helpers.tsx:34:20
    at uI (react-dom.production.min.js:244:322)
    at oU (react-dom.production.min.js:286:88)
    at react-dom.production.min.js:283:386
    at oI (react-dom.production.min.js:282:143)
    at oC (react-dom.production.min.js:273:439)
    at r4 (react-dom.production.min.js:127:100)
    at react-dom.production.min.js:267:262
(anonymous) @ instrument.js:109
fire @ maplibre-gl.js:31
queryRenderedFeatures @ maplibre-gl.js:35
queryRenderedFeatures @ maplibre-gl.js:35
T @ useFeatureMarker.tsx:21
(anonymous) @ useFeatureMarker.tsx:36
(anonymous) @ helpers.tsx:34
uI @ react-dom.production.min.js:244
oU @ react-dom.production.min.js:286
(anonymous) @ react-dom.production.min.js:283
oI @ react-dom.production.min.js:282
oC @ react-dom.production.min.js:273
r4 @ react-dom.production.min.js:127
(anonymous) @ react-dom.production.min.js:267
Show 11 more frames
Show less
helpers.ts:62 WARNING: Too many active WebGL contexts. Oldest context will be lost.
C @ helpers.ts:62
e7 @ BrowserMap.tsx:42
ab @ react-dom.production.min.js:167
ud @ react-dom.production.min.js:197
i @ react-dom.production.min.js:292
oO @ react-dom.production.min.js:280
(anonymous) @ react-dom.production.min.js:280
oS @ react-dom.production.min.js:280
x @ scheduler.production.min.js:13
T @ scheduler.production.min.js:14
Show 8 more frames
Show less
tslib.es6.js:1 [Violation] 'webglcontextrestored' handler took 164ms
scheduler.production.min.js:1 [Violation] 'message' handler took 264ms
```
</details>

--- a/src/components/Map/BrowserMap.tsx
+++ b/src/components/Map/BrowserMap.tsx
@@ -14,7 +14,7 @@ import { useUpdateStyle } from './behaviour/useUpdateStyle';
 import { useInitMap } from './behaviour/useInitMap';
 import { Translation } from '../../services/intl';
 import { useToggleTerrainControl } from './behaviour/useToggleTerrainControl';
-import { isWebglSupported } from './helpers';
+import { webglSupported } from './helpers';
 import { useOnMapLongPressed } from './behaviour/useOnMapLongPressed';
 import { useAddTopRightControls } from './useAddTopRightControls';
 
@@ -39,7 +39,7 @@ const NotSupportedMessage = () => (
 // TODO https://cdn.klokantech.com/openmaptiles-language/v1.0/openmaptiles-language.js + use localized name in FeaturePanel
 
 const BrowserMap = ({ onMapLoaded }) => {
-  if (!isWebglSupported()) {
+  if (!webglSupported) {
     onMapLoaded();
     return <NotSupportedMessage />;
   }

--- a/src/components/Map/helpers.ts
+++ b/src/components/Map/helpers.ts
@@ -1,6 +1,7 @@
 import { OsmApiId } from '../../services/helpers';
 import { basicStyle } from './styles/basicStyle';
 import { climbingLayers } from './styles/layers/climbingLayers';
+import { isBrowser } from '../helpers';
 
 const isOsmLayer = (id) => {
   if (id.startsWith('place-country-')) return false; // https://github.com/zbycz/osmapp/issues/35
@@ -51,7 +52,7 @@ export const convertOsmIdToMapId = (apiId: OsmApiId) => {
 
 // maplibregl.supported() no longer exists
 // copied from https://maplibre.org/maplibre-gl-js/docs/examples/check-for-support/
-export const isWebglSupported = () => {
+const isWebglSupported = () => {
   if (window.WebGLRenderingContext) {
     const canvas = document.createElement('canvas');
     try {
@@ -71,3 +72,5 @@ export const isWebglSupported = () => {
   // WebGL not supported
   return false;
 };
+
+export const webglSupported = isBrowser() ? isWebglSupported() : true;

